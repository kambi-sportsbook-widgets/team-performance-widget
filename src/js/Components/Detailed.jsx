import React from 'react';
import ParticipantResults from './ParticipantResults';
import EventResult from './EventResult';

const Detailed = ({ participants }) => {
   return (
      <div className='kw-detailed'>
         {
            participants.map(participant =>
               <ParticipantResults key={participant.id} name={participant.name}>
                  {
                     participant.lastEvents.map(event =>
                        <EventResult
                           key={event.start}
                           homeName={event.homeName}
                           homeScore={event.homeScore}
                           awayName={event.awayName}
                           awayScore={event.awayScore}
                        />
                     )
                  }
               </ParticipantResults>
            )
         }
      </div>
   );
};

Detailed.propTypes = {
   /**
    * The participants
    */
   participants: React.PropTypes.arrayOf(React.PropTypes.object).isRequired
};

export default Detailed;
