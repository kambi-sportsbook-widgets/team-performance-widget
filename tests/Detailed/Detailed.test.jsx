/* eslint-env jest */
import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import Detailed from '../../src/js/Components/Detailed';
import mockData from '../mockData.json';

let renderer;

describe('Detailed DOM rendering', () => {
   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
   });
   it('renders correctly for mock data', () => {
      expect(renderer.render(
         <Detailed participants={mockData} />
      )).toMatchSnapshot();
   });
});
