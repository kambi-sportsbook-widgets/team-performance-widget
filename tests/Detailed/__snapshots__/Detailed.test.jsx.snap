// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detailed DOM rendering renders correctly for mock data 1`] = `
<div
  className="kw-detailed"
>
  <ParticipantResults
    name="FC Augsburg"
  >
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="lose"
      />
      <EventResult
        awayName="FC Augsburg"
        awayScore={0}
        homeName="1 FSV Mainz 05"
        homeScore={2}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="win"
      />
      <EventResult
        awayName="Werder Bremen"
        awayScore={2}
        homeName="FC Augsburg"
        homeScore={3}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="win"
      />
      <EventResult
        awayName="FC Augsburg"
        awayScore={2}
        homeName="VfL Wolfsburg"
        homeScore={1}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="lose"
      />
      <EventResult
        awayName="1899 Hoffenheim"
        awayScore={2}
        homeName="FC Augsburg"
        homeScore={0}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="draw"
      />
      <EventResult
        awayName="FC Augsburg"
        awayScore={1}
        homeName="Hannover 96"
        homeScore={1}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="draw"
      />
      <EventResult
        awayName="AZ Alkmaar"
        awayScore={1}
        homeName="FC Augsburg"
        homeScore={1}
      />
    </div>
  </ParticipantResults>
  <ParticipantResults
    name="Bayer Leverkusen"
  >
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="win"
      />
      <EventResult
        awayName="Eintracht Frankfurt"
        awayScore={0}
        homeName="Bayer Leverkusen"
        homeScore={3}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="lose"
      />
      <EventResult
        awayName="Bayer Leverkusen"
        awayScore={0}
        homeName="Hamburger SV"
        homeScore={1}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="lose"
      />
      <EventResult
        awayName="Monchengladbach"
        awayScore={3}
        homeName="Bayer Leverkusen"
        homeScore={2}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="win"
      />
      <EventResult
        awayName="Hertha BSC"
        awayScore={1}
        homeName="Bayer Leverkusen"
        homeScore={3}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="win"
      />
      <EventResult
        awayName="VfL Bochum"
        awayScore={1}
        homeName="Bayer Leverkusen"
        homeScore={2}
      />
    </div>
    <div
      className="kw-eventresult-container"
    >
      <EventResultIndicator
        result="win"
      />
      <EventResult
        awayName="Atletico Mineiro MG"
        awayScore={0}
        homeName="Bayer Leverkusen"
        homeScore={1}
      />
    </div>
  </ParticipantResults>
</div>
`;
